File: src-tauri/src/model/invoice.rs
use crate::model::customer::Customer;
use crate::model::item::Item;
use crate::model::payment::Payment;
use crate::model::vehicle::Vehicle;

pub struct Invoice
{
    invoiceId: u32,
    dateOrdered: String,
    total: f32,
    vehicle: Vehicle,
    paymentType: Payment,
    customer: Customer,
    invoiceItems: Vec<Item>,
}

File: src/entity/Invoice.ts
import {Vehicle} from "./Vehicle.ts";
import {PaymentType} from "./PaymentType.ts";
import {Customer} from "./Customer.ts";
import {InvoiceItems} from "./InvoiceItems.ts";

export interface Invoice {
    id:number
    dateOrdered:Date,
    total:number,
    vehicle:Vehicle,
    paymentType:PaymentType
    customer:Customer
    status:string
    invoiceItems:InvoiceItems[]
}
